internal/slicer/slicer_test.go

```diff
@@ -1478,6 +1478,211 @@ var slicerTests = []slicerTest{{
 		`,
 	},
 	error: `cannot find package "test-package" in archive\(s\)`,
+}, {
+	summary: "Valid hard link in two slices in the same package",
+	slices: []setup.SliceKey{
+		{"test-package", "slice1"},
+		{"test-package", "slice2"}},
+	pkgs: []*testutil.TestPackage{{
+		Name: "test-package",
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Dir(0755, "./dir/"),
+			testutil.Reg(0644, "./dir/file", "text for file"),
+			testutil.Hln(0644, "./hardlink", "./dir/file"),
+		}),
+	}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+		package: test-package
+		slices:
+			slice1:
+				contents:
+					/dir/file:
+					/hardlink:
+			slice2:
+				contents:
+					/dir/file:
+					/hardlink:
+	`,
+	},
+	filesystem: map[string]string{
+		"/dir/":     "dir 0755",
+		"/dir/file": "file 0644 28121945",
+		"/hardlink": "file 0644 28121945",
+	},
+	manifestPaths: map[string]string{
+		"/dir/file": "file 0644 28121945 <1> {test-package_slice1,test-package_slice2}",
+		"/hardlink": "file 0644 28121945 <1> {test-package_slice1,test-package_slice2}",
+	},
+}, {
+	summary: "Empty hard link is inflated with its counterpart",
+	slices: []setup.SliceKey{
+		{"test-package", "myslice"}},
+	pkgs: []*testutil.TestPackage{{
+		Name: "test-package",
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Dir(0755, "./dir/"),
+			testutil.Reg(0644, "./dir/file", "text for file"),
+			testutil.Hln(0644, "./hardlink1", "./dir/file"),
+			testutil.Hln(0644, "./hardlink2", "./dir/file"),
+		}),
+	}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/hardlink1:
+						/hardlink2:
+		`,
+	},
+	filesystem: map[string]string{
+		"/hardlink1": "file 0644 28121945",
+		"/hardlink2": "file 0644 28121945",
+	},
+	manifestPaths: map[string]string{
+		"/hardlink1": "file 0644 28121945 <1> {test-package_myslice}",
+		"/hardlink2": "file 0644 28121945 <1> {test-package_myslice}",
+	},
+}, {
+	summary: "Single hard link has 0 hardlink identifier",
```

Comment:
Note: this should have no identifiers if we decide to not have the `<0>` identifier for regular files.